### PR TITLE
Add firebase login route

### LINKED ISSUE #38: [Login To User Account] - Implement the functionality to login using firebase
## Description

We need a way to login to our app using firebase. Here is some [documentation](https://firebase.google.com/docs/auth/web/start) on how to setup authentication on frontend. First we need to initialize a firebase app on our frontend. Then we can use the built in signInWithEmailAndPassword function that firebase provides us. Once a user is signed in they need to be routed to /home (or the student table page). Here is a [video](https://www.youtube.com/watch?v=lQftwBTCejE) I found that shows an example of how to login using firebase, they use the app router instead of the pages router but I think it gives a general sense of what needs to be done. Lastly, we need to ensure that all routes that are called from the app are protected with the auth token (add the auth bearer token to the header of each route).

## Requirments

- [ ] Firebase is initialized and login function is implemented
- [ ] User is routed to the home page
- [ ] All API calls are protected with the auth token
- [ ] All other pages cannot be accessed without having signed in

### PR DESCRIPTION
**Pull request by vs2961**
## Tracking Info

Resolves #38

## Changes

Added route for login

## Testing

Tested it with Postman.

Call `localhost:4000/api/user/login` with

```
{
    "email" : "test@testing.com",
    "password":"abc123"
}
```

## Confirmation of Change

Run the backend and run with postman.

### CODE REVIEW

**Comment by vs2961** [comment]
Weird, Parth is the default codeowner now, might want to look into that.

File: backend/src/controllers/user.ts
Comment (adhi0331) [inline backend/src/controllers/user.ts]: I would suggest also sending account status (approved or waiting for approval) since after logging in, the page they get directed to is based on whether they've been approved or not.

File: backend/src/controllers/user.ts
Comment (adhi0331) [inline backend/src/controllers/user.ts:72]: Do we want to consider encrypting the password so that it's protected? I'm sure there some basic packages we could use for encryption/decryption.

**Review by lisasiliu — APPROVED** [review]
looks good!

**Review by mraysu — APPROVED** [review]
Looks good

File: backend/src/controllers/user.ts
Comment (vs2961) [reply backend/src/controllers/user.ts]: Done

**Review by adhi0331 — APPROVED** [review]
LGTM!!! 🚀

**Pull request merged**
